modules/serverpod_auth/serverpod_auth_server/lib/src/firebase/auth/token_verifier.dart

```diff
@@ -0,0 +1,71 @@
+// https://github.com/appsup-dart/firebase_admin/blob/master/lib/src/auth/token_verifier.dart Licenced under Apache License.
+
+// Copyright (c) 2020, Rik Bellens.
+
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+
+//        http://www.apache.org/licenses/LICENSE-2.0
+
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+import 'package:openid_client/openid_client_io.dart';
+import 'package:serverpod_auth_server/src/firebase/exceptions/firebase_exception.dart';
+import 'package:http/http.dart' as http;
+
+/// JWT verifier
+class TokenVerifier {
+  final String _projectId;
+  final http.Client? _httpClient;
+
+  /// Creates a new [TokenVerifier] object with a [projectId] and [Client]
+  TokenVerifier({
+    required String projectId,
+    http.Client? httpClient,
+  })  : _projectId = projectId,
+        _httpClient = httpClient;
+
+  /// Verifies the format and signature of a Firebase Auth JWT token.
+  Future<IdToken> verifyJwt(String jwtToken) async {
+    var client = await _getOpenIdClient();
+
+    var credential = client.createCredential(idToken: jwtToken);
+
+    await for (var e in credential.validateToken()) {
+      throw FirebaseJWTException(
```

Comment:
Suggestion: There should be different exceptions for if the validation fails and if the claims fail.

That makes it possible to handle the different failures on the outside and provide better error messages.